contracts/ReversibleICO.sol

```diff
@@ -250,27 +252,27 @@ contract ReversibleICO is IERC777Recipient {
 
 
         // Setup stage 1 to n: The buy phase stages
-        uint256 lastStageBlockEnd = stage0.endBlock;
+        // Each new stage starts after the previous phase's endBlock
+        uint256 newStageStartBlock = stage0.endBlock + 1;
 
         for (uint8 i = 1; i <= _stageCount; i++) {
 
             Stage storage stageN = stages[stageCount];
             // stageCount = n
-            // Each new stage starts after the previous phase's endBlock
-            stageN.startBlock = lastStageBlockEnd + 1;
-            stageN.endBlock = lastStageBlockEnd + _stageBlockCount + 1;
+            stageN.startBlock = newStageStartBlock;
+            stageN.endBlock = newStageStartBlock + _stageBlockCount;
             // At each stage the token price increases by _stagePriceIncrease * stageCount
             stageN.tokenPrice = _commitPhasePrice + (_stagePriceIncrease * (i));
             stageCount++;
 
-            lastStageBlockEnd = stageN.endBlock;
+            newStageStartBlock = stageN.endBlock + 1;
         }
 
         // The buy phase starts on the subsequent block of the commitPhase's (stage0) endBlock
         buyPhaseStartBlock = commitPhaseEndBlock + 1;
-        buyPhaseEndBlock = lastStageBlockEnd;
+        buyPhaseEndBlock = newStageStartBlock - 1;
         // The duration of buyPhase in blocks
-        buyPhaseBlockCount = lastStageBlockEnd - buyPhaseStartBlock;
+        buyPhaseBlockCount = newStageStartBlock - buyPhaseStartBlock;
```

Comment:
Thats better because now its the correct range size of the block. @mickys you agree?
Before it was one block to less.